tcf_website/static/schedule/parse_time.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-unused-vars */ // Since these functions are called by other files
+function stringTimeToInt(stringTime) {
+  let ret = 0;
+  const parsedTime = stringTime.split(":");
+  ret += 60 * parseInt(parsedTime[0]);
+  if (parsedTime[1][2] === "p" && parsedTime[0] !== "12") {
+    ret += 720;
+  }
+  const minuteString = parsedTime[1].substr(0, 2);
+  ret += parseInt(minuteString);
+
+  return ret;
+}
+
+function parseTime(timeString) {
+  timeString = timeString.split(","); // split lab times / discussion times / lecture times
+  const weekdayMeetingTimes = Array.from({ length: 5 }, () => []); // index 0 = monday, index 1 = tuesday, index 2 = wednesday, index 3 = thursday, index 4 = friday
+
+  for (let j = 0; j < timeString.length; j++) {
```

Comment:
Maxed out spaghetti.

Split the array by uppercase letters to get each part, then map the abbreviation to the appropriate day of the week in an enum.